staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go

```diff
@@ -395,8 +395,8 @@ func (obj InfoObject) Merged() (runtime.Object, error) {
 		ResourceVersion: resourceVersion,
 	}
 
-	_, result, err := patcher.Patch(obj.Info.Object, modified, obj.Info.Source, obj.Info.Namespace, obj.Info.Name, obj.ErrOut)
-	return result, err
+	_, err = patcher.Patch(obj.Info, modified, obj.ErrOut)
+	return obj.Info.Object, err
```

Comment:
The old code did not modify obj.Info. Now obj.Info is mutated here. Does this matter? At the very least the semantics for calling `Merged()` multiple times in a row have definitely changed - this may not matter but it is a subtle change worth mentioning.